xml-conduit/src/Text/XML/Stream/Parse.hs

```diff
@@ -476,14 +478,21 @@ conduitToken :: MonadThrow m => ParseSettings -> ConduitT T.Text (PositionRange,
 conduitToken = conduitParser . parseToken
 
 parseToken :: ParseSettings -> Parser Token
-parseToken settings = (char '<' >> parseLt) <|> TokenContent <$> parseContent settings False False
```

Comment:
Do I understand correctly, that the aim of replacing `<|>` with `case` expressions, is to ensure `<?>` reports the correct information in case of failure ? Is this what you meant by `Parsers have been rewritten to ensure
that failures occur where they should` ?

Reply:
I suspect that the case in `parseLt` is also slightly more performant: instead of testing for `?`, then `!`, then `/`, and only then for a name, we can just go right away to the right branch.  But I haven't rigorously measured this, so maybe it's not.